src/Umbraco.Core/PropertyEditors/IntegerPropertyEditor.cs

```diff
@@ -50,5 +52,65 @@ public IntegerPropertyValueEditor(
                 : int.TryParse(value?.ToString(), CultureInfo.InvariantCulture, out var parsedIntegerValue)
                     ? parsedIntegerValue
                     : null;
+
+        internal abstract class IntegerPropertyConfigurationValidatorBase : ConfigurationValidatorBase
+        {
+            protected static bool TryParsePropertyValue(object? value, out int parsedIntegerValue)
```

Comment:
```suggestion
        internal abstract class IntegerPropertyConfigurationValidatorBase<TValue> : ConfigurationValidatorBase
        {
            protected abstract bool TryParsePropertyValue(object? value, out TValue parsedValue)
         }
```
I like this pattern, we could extract it to a `SimplePropertyConfigurationValidatorBase` or something for properties that require parsing/casting, however, it would only serve to ensure a more consistent pattern, so it might not be worth it, but worth considering :) Could see myself doing something very similar for boolean/string values, but just with casting.

Reply:
Yes, I think that's nice.  I'll apply.